test/e2e/expirable-airdrop.spec.ts

```diff
@@ -0,0 +1,302 @@
+  });
+
+  when('claimAndTransfer()', () => {
+    let leaf: any;
+    let proof: any;
+
+    given(async () => {
+      await prepareAirdrop();
+
+      leaf = getLeaf(bob.address, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+    });
+
+    when('airdrop is claimed by claimee', () => {
+      let tx: Transaction;
+
+      given(async () => {
+        tx = await expirableAirdrop.connect(bob).claimAndTransfer(randomAddress, airdropAmountBN, proof);
+      });
+
+      then('tokens are transferred to claimee', async () => {
+        // expect
+        expect(await token.balanceOf(randomAddress)).equal(airdropAmountBN);
+      });
+
+      then('event is emitted', async () => {
+        // expect
+        expect(tx).to.have.emit(expirableAirdrop, 'Claimed').withArgs(bob.address, randomAddress, airdropAmountBN);
+      });
+
+      then('claimee is marked as claimed', async () => {
+        // expect
+        await expect(expirableAirdrop.connect(bob).claimAndTransfer(alice.address, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+          expirableAirdrop,
+          'AlreadyClaimed'
+        );
+      });
+    });
+
+    then('revert if no airdrop', async () => {
+      leaf = getLeaf(randomAddress, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+
+      await expect(expirableAirdrop.connect(random).claimAndTransfer(randomAddress, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+        expirableAirdrop,
+        'NotInMerkle'
+      );
+    });
+
+    then('revert if expired', async () => {
+      leaf = getLeaf(carl.address, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+
+      // time travelling
+      await time.increase(3 * oneMonth);
+
+      // expect
+      await expect(expirableAirdrop.connect(carl).claimAndTransfer(randomAddress, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+        expirableAirdrop,
+        'Expired'
+      );
+    });
+  });
+
+  when('retrieveUnclaimedTokens()', () => {
+    let balanceBefore: BigNumber;
+
+    given(async () => {
+      await prepareAirdrop();
+      balanceBefore = await token.balanceOf(expirableAirdrop.address);
+    });
+
+    then('revert if not expired yet', async () => {
+      // try to retrieve
+      await expect(expirableAirdrop.retrieveUnclaimedTokens()).to.be.revertedWithCustomError(expirableAirdrop, 'NotExpired');
+    });
+
+    when('tokens back in governor wallet', () => {
+      let tx: Transaction;
+
+      given(async () => {
+        // time travelling
+        await time.increase(3 * oneMonth);
+
+        tx = await expirableAirdrop.retrieveUnclaimedTokens();
+      });
+
+      then('contract balance before retrieving is more than zero', async () => {
+        // expect
+        expect(Number(balanceBefore)).greaterThan(0);
+      });
+
+      then('tokens are transferred to governance', async () => {
+        // expect
+        expect(await token.balanceOf(deployer.address)).equal(balanceBefore);
+      });
```

Comment:
There is no need of testing the contract balance before, but you might use the `balanceBefore` to compare with the `balanceAfter`.
For example:
- ContractsBalanceNow = contractBalanceBefore + deployerBalanceNow

Kinda testing the sanity between balances of the contract and the one receiving the unclaimed airdrop.